rustup-init.sh

```diff
@@ -400,30 +402,50 @@ downloader() {
         get_ciphersuites_for_curl
         _ciphersuites="$RETVAL"
         if [ -n "$_ciphersuites" ]; then
-            curl --proto '=https' --tlsv1.2 --ciphers "$_ciphersuites" --silent --show-error --fail --location "$1" --output "$2"
+            _err=$(curl --proto '=https' --tlsv1.2 --ciphers "$_ciphersuites" --silent --show-error --fail --location "$1" --output "$2" 2>&1)
```

Comment:
this should use the `-w` option to control the output and get the http_code written in machine parseable form. See man curl.

Reply:
Yes, but we still want to show the original error in human readable form in case it wasn't 404.

Reply:
You can include other components in the output with line breaks or whatever other framing you wish.

Reply:
I don't feel like adding some parsing from command messages, I am not sure any way to do this easily. @rbtcollins you could add that in another pull request if you want.

Reply:
Given the way the PR is set up, @rbtcollins ought to be able to push an example/mock-up to your branch for the PR, if he has time and energy.

Reply:
I'm worried that what you're doing here is going to break in the future as the output isn't a stable format; curl has a stable format which is the -w format, grepping the non formatted output is just hoping that it happens to work on all curl versions everywhere.

Reply:
It's just grepping for 404, also I don't think it will change.